noir-projects/noir-protocol-circuits/crates/types/src/meta/mod.nr

```diff
@@ -423,13 +423,17 @@ pub comptime fn generate_serialize_to_fields(
 }
 
 /// From a quote that gets tokenized to a multiple tokens we collapse it to a single token by replacing all `.` with `_`.
-/// E.g. "self.value" -> "self_value"
+/// E.g. "self.values[0]" -> "self_values_0"
 comptime fn collapse_to_one_token(q: Quoted) -> Quoted {
     let tokens = q.tokens();
 
     let mut single_token = quote {};
     for token in tokens {
-        let new_token = if token == quote {.} { quote {_} } else { token };
+        let new_token = if ((token == quote {.}) | (token == quote {[}) | (token == quote {]})) {
+            quote {_}
+        } else {
+            token
+        };
```

Comment:
This was a bug and it got triggered by me using Serialize derivation in `noir-projects/noir-contracts/contracts/card_game_contract/src/game.nr`